### PR TITLE
Add URG simulator for CI

### PR DESCRIPTION
**Pull request by at-wat**
### Summary
- Add a simulator of the sensor to reproduce the sensor internal timing
- Test urg_stamped using simulator
  - Simple timestamp accuracy
  - Reboot on sensor state error

### Note
Project coverage is decreased since many source files, which weren't counted due to lack of the tests, are now detected by Codecov.

### CODE REVIEW

File: src/urg_sim/urg_sim.cpp [hunk 1]
```diff
@@ -0,0 +1,761 @@
...
+          this));
+  timestamp_epoch_ = boost::posix_time::microsec_clock::universal_time();
+
+  if (socket_.is_open())
+  {
+    std::cerr << "Closing" << std::endl;
+    socket_.close();
+    accept();
+  }
+  accept();
```
Comment (Taka-Kazu) [inline src/urg_sim/urg_sim.cpp:481]: Is this intended to call `accept()` twice if `socket_.is_open() == true`?

**Review by Taka-Kazu — APPROVED** [review]
LGTM. I left minor commnet.

**Comment by at-wat** [comment]
Noticed that `boost::asio::deadline_timer` is affected by system time update (e.g. by NTP).
It should be replaced by steady_timer

**Comment by at-wat** [comment]
> Noticed that `boost::asio::deadline_timer` is affected by system time update (e.g. by NTP). It should be replaced by steady_timer

Flakyness of the test was not due to system time update, but lack of timer reset on sensor reboot.
Fixing now

**Comment by at-wat** [comment]
Tests are stabilized by resetting the timers on reboot (https://github.com/seqsense/urg_stamped/pull/146/commits/94a954e03167723b622e1d76b2df7216b2c1531d)

File: src/urg_sim/urg_sim.cpp (hunk 1 above)
Comment (at-wat) [reply src/urg_sim/urg_sim.cpp]: Removed duplicated `accept`

**Pull request merged**